client/tests/test_online.py

```diff
@@ -0,0 +1,841 @@
+from featureform.online import (
+    PineconeConfig,
+    VectorType,
+    PineconeOnlineStore,
+)
+from featureform.register import Float32
+import os
+import sys
+from dotenv import load_dotenv
+import uuid
+import pandas as pd
+import numpy as np
+import pytest
+
+sys.path.insert(0, "client/src/")
+
+
+def get_pinecone_config():
+    pinecone_project_id = os.getenv("PINECONE_PROJECT_ID", "")
+    pinecone_api_key = os.getenv("PINECONE_API_KEY", "")
+    pinecone_environment = os.getenv("PINECONE_ENVIRONMENT", "")
+    return PineconeConfig(
+        project_id=pinecone_project_id,
+        api_key=pinecone_api_key,
+        environment=pinecone_environment,
+    )
+
+
+def get_random_feature_variant():
+    return str(uuid.uuid4()), str(uuid.uuid4())
+
+
+cases = [
+    pytest.param(
+        {
+            "config_fn": get_pinecone_config,
+            "online_store_class": PineconeOnlineStore,
+        },
+        id="Pinecone",
+    ),
+]
+
+
+@pytest.mark.parametrize("case", cases)
+def test_online_vector_stores(case):
+    load_dotenv(".env")
+    config = case["config_fn"]()
+    store = case["online_store_class"](config)
+    feature, variant = get_random_feature_variant()
+
+    # Test create_index
+    table = store.create_index(feature, variant, VectorType(Float32, 768, True))
+    assert table is not None
+
+    # Test insert
+    embeddings = pd.read_csv(
+        "client/tests/test_files/input_files/300_stock_headline_embeddings.csv"
+    )
+    for _, row in embeddings.iterrows():
+        vector = np.fromstring(row["headline_vector"][1:-1], sep=", ")
+        table.set(row["publisher"], vector.tolist())
+
+    expected = embeddings.tail(1)
+    actual = table.get(expected["publisher"].values[0])
+    assert actual is not None and actual == expected["publisher"].values[0]
+
+    # Test nearest
+    query_vector = [
```

Comment:
holy shit does this have to be this long lol

Reply:
I either read it from a file and deserialize or hardcode it here 🤷‍♂️  In either case, the dimensions have to match the index.